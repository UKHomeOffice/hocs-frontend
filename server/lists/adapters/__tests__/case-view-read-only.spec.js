const caseViewReadOnlyAdapter = require('../case-view-read-only');

const mockData = {
    schema: {
        title: 'MIN/1234567/18',
        fields: [
            {
                component: 'accordion',
                props: {
                    name: 'Accordion',
                    sections: [
                        {
                            title: 'titleA',
                            items: [
                                {
                                    component: 'text',
                                    validation: ['required'],
                                    props: {
                                        name: 'fieldA',
                                        label: 'My Field A'
                                    },
                                },
                                {
                                    component: 'date',
                                    validation: ['required'],
                                    props: {
                                        name: 'dateFieldA',
                                        label: 'My Date Field A'
                                    },
                                },
                                {
                                    component: 'text',
                                    validation: ['required'],
                                    props: {
                                        name: 'emptyFieldA',
                                        label: 'My Empty Field A'
                                    },
                                },
                            ],
                        },
                        {
                            title: 'titleB',
                            items: [
                                {
                                    component: 'text',
                                    validation: ['required'],
                                    props: {
                                        name: 'fieldB',
                                        label: 'My Field B'
                                    },
                                },
                                {
                                    component: 'date',
                                    validation: ['required'],
                                    props: {
                                        name: 'dateFieldB',
                                        label: 'My Date Field  B'
                                    },
                                },
                                {
                                    component: 'text',
                                    validation: ['required'],
                                    props: {
                                        name: 'emptyFieldB',
                                        label: 'My Empty Field B'
                                    },
                                },
                            ],
                        },
                        {
                            title: 'titleC',
                            items: [
                                {
                                    component: 'text',
                                    validation: ['required'],
                                    props: {
                                        name: 'fieldC',
                                        label: 'My Field C'
                                    },
                                },
                                {
                                    component: 'date',
                                    validation: ['required'],
                                    props: {
                                        name: 'dateFieldC',
                                        label: 'My Date Field C'
                                    },
                                },
                                {
                                    component: 'text',
                                    validation: ['required'],
                                    props: {
                                        name: 'emptyFieldC',
                                        label: 'My Empty Field C'
                                    },
                                },
                            ]
                        },
                    ],
                },
            },
        ],
    },
    data: {
        fieldA: 'Some Value A',
        fieldB: 'Some Value B',
        fieldC: 'Some Value C',
        dateFieldA: '2020-01-19',
        dateFieldB: '2020-01-20',
        dateFieldC: '2020-01-21',
    }
};

describe('Case-view-read-only Adapter', () => {
    it('should build a schema from the provided template', async () => {

        const result = await caseViewReadOnlyAdapter(mockData);

        expect(result).toBeDefined();
        expect(result).toMatchSnapshot();
    });
});
