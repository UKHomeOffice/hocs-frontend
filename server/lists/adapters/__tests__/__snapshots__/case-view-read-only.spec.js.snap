// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Case-view-read-only Adapter should build a schema from the provided template and have sections in alphabetical order 1`] = `
Object {
  "data": Object {
    "my_date_field": "19/01/2020",
    "my_empty_field": undefined,
    "my_field": "Some Value",
  },
  "meta": Object {},
  "schema": Object {
    "defaultActionLabel": "Submit",
    "fields": Array [
      Object {
        "component": "accordion",
        "props": Object {
          "name": "case-view",
          "sections": Array [
            Object {
              "items": Array [
                Object {
                  "component": "text",
                  "props": Object {
                    "label": "My Field",
                    "name": "my_field",
                  },
                  "validation": Array [
                    "required",
                  ],
                },
                Object {
                  "component": "date",
                  "props": Object {
                    "label": "My Date Field",
                    "name": "my_date_field",
                  },
                  "validation": Array [
                    "required",
                  ],
                },
                Object {
                  "component": "text",
                  "props": Object {
                    "label": "My Empty Field",
                    "name": "my_empty_field",
                  },
                  "validation": Array [
                    "required",
                  ],
                },
              ],
              "title": "STAGE_NAME",
            },
            Object {
              "items": Array [
                Object {
                  "component": "text",
                  "props": Object {
                    "label": "My Field",
                    "name": "my_field",
                  },
                  "validation": Array [
                    "required",
                  ],
                },
                Object {
                  "component": "date",
                  "props": Object {
                    "label": "My Date Field",
                    "name": "my_date_field",
                  },
                  "validation": Array [
                    "required",
                  ],
                },
                Object {
                  "component": "text",
                  "props": Object {
                    "label": "My Empty Field",
                    "name": "my_empty_field",
                  },
                  "validation": Array [
                    "required",
                  ],
                },
              ],
              "title": "STAGE_NAME1",
            },
            Object {
              "items": Array [
                Object {
                  "component": "text",
                  "props": Object {
                    "label": "My Field",
                    "name": "my_field",
                  },
                  "validation": Array [
                    "required",
                  ],
                },
                Object {
                  "component": "date",
                  "props": Object {
                    "label": "My Date Field",
                    "name": "my_date_field",
                  },
                  "validation": Array [
                    "required",
                  ],
                },
                Object {
                  "component": "text",
                  "props": Object {
                    "label": "My Empty Field",
                    "name": "my_empty_field",
                  },
                  "validation": Array [
                    "required",
                  ],
                },
              ],
              "title": "STAGE_NAME2",
            },
          ],
        },
        "validation": Array [],
      },
    ],
    "showPrimaryAction": false,
    "title": "MIN/1234567/18",
  },
}
`;
